Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabPublicHandler.cpp

```diff
@@ -1340,87 +1347,102 @@ namespace AzToolsFramework
-                // DOM value pointers can't be relied upon if the original DOM gets modified after pointer creation.
-                // This scope is added to limit their usage and ensure DOM is not modified when it is being used.
-                AZStd::vector<AZStd::pair<const PrefabDomValue*, AZStd::string>> removedEntityDomAndPathList;
+                const InstanceAlias instanceAlias = instance->GetInstanceAlias();
 
-                for (const AZ::Entity* entity : entityList)
-                {
-                    const AZ::EntityId entityId = entity->GetId();
-                    const AZStd::string entityAliasPath =
-                        m_instanceToTemplateInterface->GenerateEntityAliasPath(entityId); // retrieve before detaching
-
-                    // Captures the parent DOM if it is first seen in map.
-                    AZ::EntityId parentEntityId;
-                    AZ::TransformBus::EventResult(parentEntityId, entityId, &AZ::TransformBus::Events::GetParentId);
-                    CaptureInitialEntityDomFromOwningTemplate(parentEntityDomBeforeRemovalMap, parentEntityId, commonOwningTemplateDom);
-
-                    // Detaches the entity.
-                    commonOwningInstance->get().DetachEntity(entityId).release();
-                    AZ::ComponentApplicationBus::Broadcast(&AZ::ComponentApplicationRequests::DeleteEntity, entityId);
+                // Captures the parent entity that needs to be updated.
+                AZ::EntityId parentEntityId;
+                AZ::TransformBus::EventResult(parentEntityId, instance->GetContainerEntityId(), &AZ::TransformBus::Events::GetParentId);
 
-                    // Captures the entity DOM from the owning template DOM for undo purpose and puts the entity DOM into a list
-                    // for batch processing.
-                    PrefabDomPath entityDomInOwningTemplatePath(entityAliasPath.c_str());
+                if (parentEntityId.IsValid() && (entitiesThatWillBeRemoved.find(parentEntityId) == entitiesThatWillBeRemoved.end()))
+                {
+                    parentEntitysThatNeedUpdate.insert(GetEntityById(parentEntityId));
+                }
 
-                    removedEntityDomAndPathList.emplace_back(
-                        entityDomInOwningTemplatePath.Get(commonOwningTemplateDom), AZStd::move(entityAliasPath));
+                AZStd::unique_ptr<Instance> detachedInstance = commonOwningInstance->get().DetachNestedInstance(instanceAlias);
 
-                    entitiesThatGotRemoved.insert(entityId);
+                if (isOverrideEditing)
+                {
+                    detachedInstanceAliasPaths.push_back(AZStd::move(PrefabDomUtils::PathStartingWithInstances + instanceAlias));
+                }
+                else
+                {
+                    // Removes the link if it is source template editing.
+                    RemoveLink(detachedInstance, commonOwningInstance->get().GetTemplateId(), undoBatch.GetUndoBatch());
```

Comment:
Would it make sense to move this to the DeleteEntities? That way we would get a nice separation from entity/instance object manipulation vs. prefab DOM manipulation.

Reply:
Do you mean DeleteEntities helper function or delete undo node? Two of reasons would be reusing RemoveLink in handler and managing Instance lifecycle in handler instead of passing down instance pointers to helper.

Reply:
The benefit I see is the code would be easier to follow. PrefabUndoHelpers::DeleteEntitiesAndPrefabsAsOverride would be more consistent with PrefabUndoHelpers::DeleteEntities.

Reply:
We can definitely extend from the current classes. I can create a backlog task for this. For now I want to keep it simple and reuse what we already have to make sure delete-prefab override works correctly.

Once deletion in multiple owning instances and instantiate-prefab as override gets in, at that time I think we can do the refactoring for link updates.